### PR TITLE
Add support for sites deployed to sub-paths

### PR DESCRIPTION
**Pull request by jeffsee55**
Closes https://github.com/tinacms/tinacms/issues/3906

~~TODO: Check links work correctly in the admin~~

### CODE REVIEW

**Comment by jamespohalloran** [comment]
This seems to work well! Only thing, I see Jekyll & GitHub call this a "baseURL" or "basePath". https://github.com/jamespohalloran/jekyll-test/blob/main/.github/workflows/jekyll.yml#LL53C87-L53C87
Wonder if that makes more sense than subPath

**Comment by jeffsee55** [comment]
> "baseURL" or "basePath"

Hmm, yeah - maybe `basePath`?

File: .changeset/twelve-frogs-protect.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+---
+
+Add support for sites deployed to sub-paths. To enabled, provide the sub-path at config.build.subPath:
+
+```ts
+  ...
+  build: {
+    outputFolder: 'admin',
+    publicFolder: 'public',
+    basePath: 'my-site', // site is served at my-domain.com/my-site
```
Comment (kldavis4) [inline .changeset/twelve-frogs-protect.md:13]: is the key `subPath` or `basePath`?

File: .changeset/twelve-frogs-protect.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+---
+'@tinacms/schema-tools': patch
+'@tinacms/cli': patch
+---
+
+Add support for sites deployed to sub-paths. To enabled, provide the sub-path at config.build.subPath:
```
Comment (kldavis4) [inline .changeset/twelve-frogs-protect.md:6]:
```suggestion
Add support for sites deployed to sub-paths. To enabled, provide the sub-path at config.build.basePath:
```

**Review by jamespohalloran — APPROVED** [review]

**Pull request merged**